### ISSUE TITLE
100% cpu usage, polling the file tl is looking at

### ISSUE THREAD

**Issue by nealmcb** [Textualize/toolong#17]
Whenever I run `tl`, it uses 100% of a cpu. Running strace -fp <processid> shows three lines over and over, perhaps 100,000 times a second.
```
[pid 219465] lseek(7, 0, SEEK_CUR)      = 423
[pid 219465] read(7, "", 65536)         = 0
[pid 219465] poll([{fd=7, events=POLLIN}], 1, 100) = 1 ([{fd=7, revents=POLLIN}])
[pid 219465] lseek(7, 0, SEEK_CUR)      = 423
[pid 219465] read(7, "", 65536)         = 0
[pid 219465] poll([{fd=7, events=POLLIN}], 1, 100) = 1 ([{fd=7, revents=POLLIN}])
```

File descriptor 7 is the file I'm running `tl` on.  423 is the length of the file in bytes.

This is on Ubuntu 22.04.3. `tl` was installed via pipx, version 1.0.3

SInce [efficiently watching a file for changes is a feature](https://textual.textualize.io/blog/2024/02/11/file-magic-with-the-python-standard-library/#watching-a-file-for-changes), something seems broken.

**Comment by tstack** [comment]
On linux, `poll()` is used in here:

https://github.com/Textualize/toolong/blob/9fe54a6bcd6bcc2b1066c762057b0de812dcd155/src/toolong/watcher.py#L14-L15

However, `poll()` / `select()` do not work with regular files (and neither does `epoll()`, which is why it was returning `EPERM`).  These functions will always return that a file is ready to read/write.  So, the loop in this file will spin and use 100% CPU.

The code likely works on something like MacOS/FreeBSD because they use the `kqueue` selector which has support for files.

**Comment by nealmcb** [comment]
Many thanks, @tstack.
I'm sure this is overly crude, but if I don't care about breaking the tail feature, it seems I can avoid the cpu loop by simply commenting out this line:

https://github.com/Textualize/toolong/blob/9fe54a6bcd6bcc2b1066c762057b0de812dcd155/src/toolong/watcher.py#L77

A quick search suggests an approach that should be a more reasonable option:

[How can I tail a log file in Python? ... Purely pythonic solution using non-blocking readline()](https://stackoverflow.com/a/54263201/507544)

It requires polling, but the current code also seems to set a timeout and thus poll every tenth of a second.

**Comment by willmcgugan** [comment]
Thanks. Should be fixed in 1.1.1

I feel cheated by the Python docs that implied files would work everywhere.
